core/trino-main/src/main/java/io/trino/operator/PartitionedLookupSource.java

```diff
@@ -150,6 +152,70 @@ public long getJoinPosition(int position, Page hashChannelsPage, Page allChannel
         return encodePartitionedJoinPosition(partition, toIntExact(joinPosition));
     }
 
+    @Override
+    public long[] getJoinPositions(int[] positions, Page hashChannelsPage, Page allChannelsPage)
```

Comment:
previously it was accepting array, not it's returning one. Does that reduce performance?